modules/libarchive/3.7.3/patches/build_with_bazel.patch

```diff
@@ -0,0 +1,1809 @@
++#define HAVE_STDLIB_H 1
++#define HAVE_STRCHR 1
++#define HAVE_STRNLEN 1
++#define HAVE_STRDUP 1
++#define HAVE_STRERROR 1
++#define HAVE_STRFTIME 1
++#define HAVE_STRING_H 1
++#define HAVE_STRRCHR 1
++#define HAVE_SYS_STAT_H 1
++#define HAVE_SYS_TYPES_H 1
++#define HAVE_SYS_UTIME_H 1
++#define HAVE_TIME_H 1
++#define HAVE_TZSET 1
++#define HAVE_UTIME 1
++#define HAVE_WCHAR_H 1
++#define HAVE_WCHAR_T 1
++#define HAVE_WCRTOMB 1
++#define HAVE_WCSCMP 1
++#define HAVE_WCSCPY 1
++#define HAVE_WCSLEN 1
++#define HAVE_WCTOMB 1
++#define HAVE_WCTYPE_H 1
++#define HAVE_WINCRYPT_H 1
++#define HAVE_WINDOWS_H 1
++#define HAVE_WINIOCTL_H 1
++#define HAVE__CrtSetReportMode 1
++#define HAVE_CTIME_S 1
++#define HAVE__FSEEKI64 1
++#define HAVE__GET_TIMEZONE 1
++#define HAVE_GMTIME_S 1
++#define HAVE_LOCALTIME_S 1
++#define HAVE__MKGMTIME 1
++#define ICONV_CONST
++#define HAVE_LZ4HC_H 1
++
++#define LIBARCHIVE_VERSION_NUMBER "3007003"
++#define LIBARCHIVE_VERSION_STRING "3.7.3"
++#define SIZEOF_WCHAR_T 2
++#define VERSION "3.7.3"
++
++#define gid_t short
++#define id_t short
++#define mode_t unsigned short
++#define pid_t int
++#define ssize_t int64_t
++#define uid_t short
+diff --git a/libarchive_fe/BUILD.bazel b/libarchive_fe/BUILD.bazel
+new file mode 100644
+index 00000000..8776c21c
+--- /dev/null
++++ b/libarchive_fe/BUILD.bazel
+@@ -0,0 +1,18 @@
++load("@rules_cc//cc:defs.bzl", "cc_library")
++load("//bazel:config.bzl", "PLATFORM_LOCAL_DEFINES")
++
++package(default_visibility = ["//visibility:public"])
++
++cc_library(
++    name = "archive_fe",
++    hdrs = glob(["*.h"]),
++    srcs = glob(["*.c"]),
++    defines = PLATFORM_LOCAL_DEFINES,
++    deps = ["//:config"],
++    strip_include_prefix = ".",
++)
++
++alias(
++    name = "libarchive_fe",
++    actual = ":archive_fe",
++)
+diff --git a/tar/BUILD.bazel b/tar/BUILD.bazel
+new file mode 100644
+index 00000000..5068b0c3
+--- /dev/null
++++ b/tar/BUILD.bazel
+@@ -0,0 +1,33 @@
++load("@rules_cc//cc:defs.bzl", "cc_binary")
++
++package(default_visibility = ["//visibility:public"])
++
++cc_binary(
++    name = "bsdtar",
++    srcs = [
++        "bsdtar.c",
++        "bsdtar.h",
++        "bsdtar_platform.h",
++        "cmdline.c",
++        "creation_set.c",
++        "read.c",
++        "subst.c",
++        "util.c",
++        "write.c",
++    ] + select({
++        "@platforms//os:windows": [
++            "bsdtar_windows.c",
++            "bsdtar_windows.h",
++        ],
++        "//conditions:default": [],
++    }),
++    deps = [
++        "//libarchive",
++        "//libarchive_fe",
++    ],
++)
++
++alias(
++    name = "tar",
++    actual = ":bsdtar",
++)
+diff --git a/tar/bsdtar.c b/tar/bsdtar.c
+index b070e0fa..4e61734f 100644
+--- a/tar/bsdtar.c
++++ b/tar/bsdtar.c
+@@ -1075,7 +1075,11 @@ static const char *long_help_msg =
+ 	"  -c Create  -r Add/Replace  -t List  -u Update  -x Extract\n"
+ 	"Common Options:\n"
+ 	"  -b #  Use # 512-byte records per I/O block\n"
++#ifdef _PATH_DEFTAPE 
```

Comment:
yeah I have no clue how this works upstream; maybe apple patches this as well :)